modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/MysqlSchemaServerCodegen.java

```diff
@@ -0,0 +1,1028 @@
+        if (Boolean.FALSE.equals(getJsonDataTypeEnabled())) {
+            columnDefinition.put("colDataType", "TEXT");
+        }
+
+        if (Boolean.TRUE.equals(required)) {
+            columnDefinition.put("colNotNull", true);
+        } else {
+            columnDefinition.put("colNotNull", false);
+            try {
+                columnDefinition.put("colDefault", toCodegenMysqlDataTypeDefault(defaultValue, (String) columnDefinition.get("colDataType")));
+            } catch (RuntimeException exception) {
+                LOGGER.warn("Property '" + baseName + "' of model '" + model.getName() + "' mapped to MySQL data type which doesn't support default value");
+                columnDefinition.put("colDefault", null);
+            }
+        }
+
+        if (description != null) {
+            columnDefinition.put("colComment", description);
+        }
+    }
+
+    public void processModelTypeProperty(CodegenModel model, CodegenProperty property) {
+        Map<String, Object> vendorExtensions = property.getVendorExtensions();
+        Map<String, Object> mysqlSchema = new HashMap<String, Object>();
+        Map<String, Object> columnDefinition = new HashMap<String, Object>();
+        String baseName = property.getBaseName();
+        Boolean required = property.getRequired();
+        String description = property.getDescription();
+        String defaultValue = property.getDefaultValue();
+
+        if (vendorExtensions.containsKey(CODEGEN_VENDOR_EXTENSION_KEY)) {
+            // user already specified schema values
+            LOGGER.info("Found vendor extension in '" + baseName + "' property, autogeneration skipped");
+            return;
+        }
+
+        vendorExtensions.put(CODEGEN_VENDOR_EXTENSION_KEY, mysqlSchema);
+        mysqlSchema.put("columnDefinition", columnDefinition);
+        columnDefinition.put("colName", toColumnName(baseName));
+        columnDefinition.put("colDataType", "TEXT");
+
+        if (Boolean.TRUE.equals(required)) {
+            columnDefinition.put("colNotNull", true);
+        } else {
+            columnDefinition.put("colNotNull", false);
+            try {
+                columnDefinition.put("colDefault", toCodegenMysqlDataTypeDefault(defaultValue, (String) columnDefinition.get("colDataType")));
+            } catch (RuntimeException exception) {
+                LOGGER.warn("Property '" + baseName + "' of model '" + model.getName() + "' mapped to MySQL data type which doesn't support default value");
+                columnDefinition.put("colDefault", null);
+            }
+        }
+
+        if (description != null) {
+            columnDefinition.put("colComment", description);
+        }
+    }
+
+    /**
+     * Generates codegen property for MySQL data type argument
+     *
+     * @param value   argument value
+     * @param hasMore shows whether codegen has more arguments or not
+     * @return generated codegen property
+     */
```

Comment:
:+1 for the doc string

Later (not this PR), you may add doc string for other functions (e.g. `processJsonTypeProperty`) in this file.